sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/util/RateLimiter.java

```diff
@@ -15,29 +15,37 @@ public class RateLimiter implements AutoCloseable {
     private final AtomicInteger inFlight = new AtomicInteger(0);
     private final AtomicInteger bucket = new AtomicInteger(0);
     private final Timer replenishTimer;
+    private final boolean enabled;
 
     public RateLimiter(int maxRps, int maxConcurrency) {
         this.maxConcurrency = maxConcurrency;
-
-        replenishTimer = new Timer("replenish");
-        replenishTimer.schedule(new TimerTask() {
-            public void run() {
-                bucket.set(maxRps);
-            }
-        }, 0, 1000L);
+        this.enabled = maxRps > 0;
+        this.replenishTimer = new Timer("replenish");
+        if (enabled) {
+            replenishTimer.schedule(new TimerTask() {
+                public void run() {
+                    bucket.set(maxRps);
+                }
+            }, 0, 1000L);
+        }
     }
 
-    public Mono<Void> acquire() {
+    public Mono<Boolean> acquire() {
```

Comment:
I don't think we need this to return a value because it'll always return true. (Otherwise, it does this repeat loop until it does get the lock and then returns true.) I feel like Mno.empty() is enough to convey that.

Reply:
sorry, noticed your comment after merging.

Unfortunately `Mono.empty()` does not play nicely with `Mono.usingWhen` since it does not emit a signal, so :(